lib/analysis_options.yaml

```diff
@@ -1,6 +1,95 @@
+include: package:flutter_lints/flutter.yaml
+
 analyzer:
   exclude:
 
 linter:
   rules:
+    # https://dart-lang.github.io/linter/lints/index.html
+    # rules with # default are already enabled by flutter_lints but specified here for completeness
+    
+    # Error Rules
+    always_use_package_imports: true
+    avoid_dynamic_calls: true
+    avoid_empty_else: true
+    avoid_print: true # default
+    avoid_returning_null_for_future: true
+    avoid_types_as_parameter_names: true
+    avoid_web_libraries_in_flutter: true # default
+    cancel_subscriptions: true
+    close_sinks: true
+    empty_statements: true
+    literal_only_boolean_expressions: true
+    no_adjacent_strings_in_list: true
+    no_logic_in_create_state: true # default
+    prefer_relative_imports: false
+    prefer_void_to_null: true
+    unnecessary_statements: true
+    use_build_context_synchronously: true # default
+    use_key_in_widget_constructors: true # default
+    valid_regexps: true
+
+    # Style Rules
+    always_declare_return_types: true
+    always_put_control_body_on_new_line: true
+    always_put_required_named_parameters_first: true
+    always_specify_types: false
+    annotate_overrides: true
+    avoid_bool_literals_in_conditional_expressions: true
+    avoid_final_parameters: true
+    avoid_multiple_declarations_per_line: true
+    avoid_returning_null: true
+    avoid_returning_null_for_void: true
+    avoid_unnecessary_containers: true # default
+    avoid_unused_constructor_parameters: true
+    avoid_void_async: true
```

Comment:
There are times that you want an async void function.

Reply:
This rule makes sure the return type is `Future<void>` rather than just `void`. So it makes sense to have :wink: